couchdb/server/compiler.py

```diff
@@ -17,6 +17,24 @@
 log = logging.getLogger(__name__)
 
 
+class EggExports(dict):
+    """Sentinel for egg export statements."""
+
+
+def maybe_b64egg(b64str):
+    """Checks if passed string is base64 encoded egg file"""
+    # Quick and dirty check for base64 encoded zipfile.
+    # Saves time and IO operations in most cases.
+    return isinstance(b64str, util.strbase) and b64str.startswith('UEsDBBQAAAAIA')
```

Comment:
Hmm.. Actually, I can found the spec and confirm `UEsDBBQAAAAIA` is egg's magic number. I only know that an egg file is zipped; and the string `UEsDBBQAAAAIA` indicate 13*3 bytes.
The zip magic number is only 4 bytes.

And... in this patch, I just decode the `b64str[:8]`, not the whole string. Do this implementation make sense to you?